source/io/module_eggShell.js

```diff
@@ -162,38 +191,178 @@
         Module.eggShell.maxExecWakeUpTime = publicAPI.eggShell.maxExecWakeUpTime = Vireo_MaxExecWakeUpTime;
 
         Module.eggShell.reboot = publicAPI.eggShell.reboot = function () {
-            EggShell_Delete(v_userShell);
-            EggShell_Delete(v_root);
-            v_root = EggShell_Create(0);
-            v_userShell = EggShell_Create(v_root);
+            EggShell_Delete(Module.eggShell.v_userShell);
+            EggShell_Delete(Module.eggShell.v_root);
+            Module.eggShell.v_root = EggShell_Create(0);
+            Module.eggShell.v_userShell = EggShell_Create(Module.eggShell.v_root);
+        };
+
+        Module.eggShell.createValueRef = function (typeRef, dataRef) {
+            return Object.freeze({
+                typeRef: typeRef,
+                dataRef: dataRef
+            });
+        };
+
+        Module.eggShell.findValueRef = publicAPI.eggShell.findValueRef = function (vi, path) {
+            var stack = Module.stackSave();
+
+            var viStackPointer = Module.coreHelpers.writeJSStringToStack(vi);
+            var pathStackPointer = Module.coreHelpers.writeJSStringToStack(path);
+            var typeStackPointer = Module.stackAlloc(POINTER_SIZE);
+            var dataStackPointer = Module.stackAlloc(POINTER_SIZE);
+
+            var eggShellResult = Module._EggShell_FindValue(Module.eggShell.v_userShell, viStackPointer, pathStackPointer, typeStackPointer, dataStackPointer);
+            if (eggShellResult !== EGGSHELL_RESULT.SUCCESS) {
+                throw new Error('A ValueRef could not be made for the following reason: ' + eggShellResultEnum[eggShellResult] +
+                    ' (error code: ' + eggShellResult + ')' +
+                    ' (vi name: ' + vi + ')' +
+                    ' (path: ' + path + ')');
+            }
+
+            var typeRef = Module.getValue(typeStackPointer, 'i32');
+            var dataRef = Module.getValue(dataStackPointer, 'i32');
+            var valueRef = Module.eggShell.createValueRef(typeRef, dataRef);
+
+            Module.stackRestore(stack);
+            return valueRef;
+        };
+
+        Module.eggShell.findSubValueRef = publicAPI.eggShell.findSubValueRef = function (valueRef, path) {
```

Comment:
path should be renamed to subPath to reflect it is not the full path.

Reply:
I'll change it.